giskard/llm/client/copilot.py

```diff
@@ -0,0 +1,95 @@
+from typing import Optional, Sequence
+
+from dataclasses import dataclass
+from logging import warning
+
+from ..config import LLMConfigurationError
+from ..errors import LLMImportError
+from .base import ChatMessage
+from .openai import AUTH_ERROR_MESSAGE, OpenAIClient
+
+try:
+    import openai
+except ImportError as err:
+    raise LLMImportError(flavor="talk") from err
+
+
+@dataclass
+class ToolChatMessage(ChatMessage):
+    name: Optional[str] = None
+    tool_call_id: Optional[str] = None
+    tool_calls: Optional[list] = None
```

Comment:
nitpick, could be more precise: `Optional[list[ChatCompletionMessageToolCall]]`

Reply:
I'll add it, I forget if adding a typing based on a conditional imports create troubles (even if put under quotes), I'll try it.